src/opm/io/eclipse/LODSmryOutput.cpp

```diff
@@ -0,0 +1,186 @@
+
+    for (size_t n = 0; n < static_cast<size_t>(m_nVect); n++)
+        m_smrydata[n].push_back(ts_data[n]);
+
+    {
+        Opm::EclIO::EclOutput outFile(m_outputFileName, m_fmt, std::ios::out);
+
+        outFile.write<int>("START", m_start_date_vect);
+
+        if (m_restart_rootn.size() > 0) {
+            outFile.write<std::string>("RESTART", {m_restart_rootn});
+            outFile.write<int>("RSTNUM", {m_restart_step});
+        }
+
+        outFile.write("KEYCHECK", m_smry_keys);
+        outFile.write("UNITS", m_smryUnits);
+
+        outFile.write<int>("RSTEP", m_rstep);
+        outFile.write<int>("TSTEP", m_tstep);
+
+        for (size_t n = 0; n < static_cast<size_t>(m_nVect); n++ ) {
+            std::string vect_name="V" + std::to_string(n);
+            outFile.write<float>(vect_name, m_smrydata[n]);
+        }
+    }
+
+    m_nTimeSteps++;
+}
+
+
+std::vector<std::string> LODSmryOutput::make_modified_keys(const std::vector<std::string> valueKeys, const GridDims& dims)
+{
+    std::vector<std::string> mod_keys;
+    mod_keys.reserve(valueKeys.size());
+
+    for (size_t n=0; n < valueKeys.size(); n++){
+
+        if (valueKeys[n].substr(0,15) == "SMSPEC.Internal"){
+            std::string mod_key = valueKeys[n].substr(16);
+            int p = mod_key.find_first_of(".");
+            mod_key=mod_key.substr(0,p);
+            mod_keys.push_back(mod_key);
+
+        } else if (valueKeys[n].substr(0,1) == "C"){
+            size_t p = valueKeys[n].find_first_of(":");
+            p = valueKeys[n].find_first_of(":", p + 1);
+
+            int num = std::stod(valueKeys[n].substr(p + 1)) - 1;
+
+            auto ijk = ijk_from_global_index(dims, num);
+
+            std::string mod_key = valueKeys[n].substr(0, p + 1);
+            mod_key = mod_key + std::to_string(ijk[0] + 1) + "," + std::to_string(ijk[1] + 1) + "," + std::to_string(ijk[2] + 1);
+
+            mod_keys.push_back(mod_key);
+
+        } else if (valueKeys[n].substr(0,1) == "B"){
+
+            size_t p = valueKeys[n].find_first_of(":");
+
+            int num = std::stod(valueKeys[n].substr(p + 1)) - 1;
+
+            auto ijk = ijk_from_global_index(dims, num);
+
+            std::string mod_key = valueKeys[n].substr(0, p + 1);
+            mod_key = mod_key + std::to_string(ijk[0] + 1) + "," + std::to_string(ijk[1] + 1) + "," + std::to_string(ijk[2] + 1);
+
+            mod_keys.push_back(mod_key);
+
+        } else {
+            mod_keys.push_back(valueKeys[n]);
+        }
+    }
+
+    return mod_keys;
+
+}
+
+std::array<int, 3> LODSmryOutput::ijk_from_global_index(const GridDims& dims, int globInd) const
+{
+
+    if (globInd < 0 || static_cast<size_t>(globInd) >= dims[0] * dims[1] * dims[2])
+        throw std::invalid_argument("global index out of range");
+
+    std::array<int, 3> result;
+    result[2] = globInd / (dims[0] * dims[1]);
+
+    int rest = globInd % (dims[0] * dims[1]);
+
+    result[1] = rest / dims[0];
+    result[0] = rest % dims[0];
+
```

Comment:
If you write this in the reverse order you get the same result with fewer arithmetic operations
```C++
result[0] = globInd % dims[0];  globInd /= dims[0];
result[1] = globInd % dims[1];
result[2] = globInd / dims[1];
```